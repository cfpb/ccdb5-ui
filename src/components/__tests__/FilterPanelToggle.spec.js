import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import ReduxFilterPanelToggle, {
  FilterPanelToggle,
  mapDispatchToProps,
  mapStateToProps,
} from '../Filters/FilterPanelToggle';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

/**
 *
 */
function setupEnzyme() {
  const props = {
    aggs: {},
    onFilterToggle: jest.fn(),
    hasButton: true,
    hasFilterToggle: true,
    hasFilters: false,
  };

  const target = shallow(<FilterPanelToggle {...props} />);

  return {
    props,
    target,
  };
}

/**
 *
 * @param hasFilters
 */
function setupSnapshot(hasFilters) {
  const middlewares = [thunk];
  const mockStore = configureMockStore(middlewares);
  const store = mockStore({
    view: {
      hasFilters,
    },
  });

  return renderer.create(
    <Provider store={store}>
      <ReduxFilterPanelToggle />
    </Provider>
  );
}

describe('initial state', () => {
  it('renders Close filters without crashing', () => {
    const target = setupSnapshot(true);
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Filter Results without crashing', () => {
    const target = setupSnapshot(false);
    const tree = target.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('mapDispatchToProps', () => {
  it('hooks into onFilterToggle', () => {
    const dispatch = jest.fn();
    mapDispatchToProps(dispatch).onFilterToggle();
    expect(dispatch.mock.calls.length).toEqual(1);
  });

  it('allows the user to trigger Filter Panel', () => {
    const { target, props } = setupEnzyme();
    const button = target.find('button');

    button.simulate('click');
    expect(props.onFilterToggle).toHaveBeenCalled();
  });
});

describe('mapStateToProps', () => {
  it('maps state and props', () => {
    const state = {
      view: {
        hasFilters: true,
      },
    };
    const actual = mapStateToProps(state);
    expect(actual).toEqual({
      hasFilters: true,
    });
  });
});
