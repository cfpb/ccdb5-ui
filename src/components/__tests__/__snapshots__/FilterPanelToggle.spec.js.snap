// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial state renders Close filters without crashing 1`] = `
<section
  className="filter-panel-toggle"
>
  <div
    className="m-btn-group"
  >
    <p>
       
    </p>
    <button
      className="a-btn"
      onClick={[Function]}
    >
      Close Filters
    </button>
  </div>
</section>
`;

exports[`initial state renders Filter Results without crashing 1`] = `
<section
  className="filter-panel-toggle"
>
  <div
    className="m-btn-group"
  >
    <p>
       
    </p>
    <button
      className="a-btn"
      onClick={[Function]}
    >
      Filter results
    </button>
  </div>
</section>
`;
